### PR TITLE
feat: Add toString() methods to errors

### PR DESCRIPTION
**Pull request by martinalbert**
## Description

Add `toString()` methods to error classes that don't contain this method.

## Motivation and Context

Error formatting.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](SECURITY.md) is correct.
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/internal/interpreter/map-interpreter.errors.ts [hunk 1]
```diff
@@ -99,6 +99,29 @@ export class HTTPError extends MapInterpreterErrorBase {
...
+  override toString(): string {
+    return [
+      `${this.kind}: ${this.message}`,
+      this.astPath ? `AST Path: ${this.astPath.join('.')}` : undefined,
+      this.metadata?.node?.location
+        ? `Original Map Location: Line ${this.metadata.node.location.line}, column ${this.metadata.node.location.column}`
+        : undefined,
+      this.request?.url ? `Request URL: ${this.request.url}` : undefined,
+      this.request?.headers
+        ? `Request headers:\n${Object.entries(this.request.headers)
```
Comment (lukas-valenta) [inline src/internal/interpreter/map-interpreter.errors.ts:112]: i wouldn't put this here, headers can contain sensitive information

Comment (martinalbert) [reply src/internal/interpreter/map-interpreter.errors.ts:112]: make sense 👍🏻

Comment (martinalbert) [reply src/internal/interpreter/map-interpreter.errors.ts:112]: What about response headers though?

Comment (lukas-valenta) [reply src/internal/interpreter/map-interpreter.errors.ts:112]: same. there's sensitive debug for this

Comment (martinalbert) [reply src/internal/interpreter/map-interpreter.errors.ts]: ok thanks, removed 🙌🏻

**Review by lukas-valenta — APPROVED** [review]

File: src/internal/interpreter/map-interpreter.errors.ts (hunk 1 above)
Comment (TheEdward162) [reply src/internal/interpreter/map-interpreter.errors.ts]: Doesn't the url also contain sensitive info? Or is this the url before query parameters would be resolved?

Comment (lukas-valenta) [reply src/internal/interpreter/map-interpreter.errors.ts]: good point

**Review by TheEdward162 — APPROVED** [review]

**Pull request merged**